osbs/http.py

```diff
@@ -131,7 +178,12 @@ def __init__(self, url, method, data=None, kerberos_auth=False,
             args['stream'] = True
 
         args['headers'] = headers
-        self.req = requests.request(method, url, **args)
+
+        if retries_disabled:
+            request_method = requests.request
+        else:
+            request_method = self.session.request
```

Comment:
Right, sessions gives us a connection pool, so its a good idea to rewrite this to:
```
if not retries_disabled:
    self.session.mount('http://', HTTPAdapter(max_retries=retry))
```
and use `self.session` always